sql-plugin/src/main/scala/com/nvidia/spark/rapids/AlluxioUtils.scala

```diff
@@ -16,13 +16,36 @@
 
 package com.nvidia.spark.rapids
 
+import java.io.FileNotFoundException
+
 import org.apache.hadoop.fs.Path
 
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.SparkSession
 import org.apache.spark.sql.catalyst.expressions.{Expression, PlanExpression}
 import org.apache.spark.sql.execution.datasources.{FileIndex, HadoopFsRelation, InMemoryFileIndex}
 import org.apache.spark.sql.execution.datasources.rapids.GpuPartitioningUtils
 
-object AlluxioUtils {
+
+object AlluxioUtils extends Logging {
+  val checkedAlluxioPath = scala.collection.mutable.HashSet[String]()
+  def checkAlluxioMounted(sparkSession: SparkSession, alluxio_path: String): Unit = {
+    this.synchronized {
+      if (!checkedAlluxioPath.contains(alluxio_path)) {
+        val path = new Path(alluxio_path)
+        val fs = path.getFileSystem(sparkSession.sparkContext.hadoopConfiguration)
+        if (!fs.exists(path)) {
+          throw new FileNotFoundException(
+            s"Alluxio path $alluxio_path does not exist, maybe forgot to mount it")
+        }
+        logInfo(s"Alluxio path $alluxio_path is mounted")
+        checkedAlluxioPath.add(alluxio_path)
```

Comment:
Should we add some limitation on this set size? Since it keeps alive as long as the process.

Reply:
Since I only check the root paths now, maybe we can just remove this global variable and check every time.
The root paths should not be many.

Reply:
Or whether it can be cleared when no longer needed.